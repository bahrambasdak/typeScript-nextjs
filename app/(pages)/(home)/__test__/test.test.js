import { render } from '@testing-library/react'
import Page from '@/app/(pages)/(home)/page'

 
it('renders homepage unchanged', () => {
  const { container } = render(<Page />)
  expect(container).toMatchSnapshot()
})