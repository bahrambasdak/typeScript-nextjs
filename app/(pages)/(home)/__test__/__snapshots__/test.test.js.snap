// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="flex w-full px-10  h-12 justify-around items-center  gap-4 transition-all"
  >
    <button
      class="h-full bg-gray-50 rounded-md grow transition-all"
      style="background-color: gray; color: black;"
    >
      <div
        class="transition-all flex  w-fit items-center justify-center group/letter "
        style="width: 100%; height: 100%;"
      >
        <div
          class="h-[1em] relative overflow-hidden"
        >
          <div
            class="h-[1em] opacity-0"
          >
            tab 1
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
          >
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              1
            </span>
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
          >
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              1
            </span>
          </div>
        </div>
      </div>
    </button>
    <button
      class="h-full bg-gray-50 rounded-md grow transition-all"
      style="background-color: gray; color: black;"
    >
      <div
        class="transition-all flex  w-fit items-center justify-center group/letter "
        style="width: 100%; height: 100%;"
      >
        <div
          class="h-[1em] relative overflow-hidden"
        >
          <div
            class="h-[1em] opacity-0"
          >
            tab 2
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
          >
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              2
            </span>
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
          >
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              2
            </span>
          </div>
        </div>
      </div>
    </button>
    <button
      class="p-4 bg-gray-50 rounded-md grow transition-all"
      style="background-color: gray; color: black;"
    >
      <div
        class="transition-all flex  w-fit items-center justify-center group/letter "
        style="width: 100%; height: 100%;"
      >
        <div
          class="h-[1em] relative overflow-hidden"
        >
          <div
            class="h-[1em] opacity-0"
          >
            tab 3
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
          >
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              3
            </span>
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
          >
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              3
            </span>
          </div>
        </div>
      </div>
    </button>
    <button
      class="p-4 bg-gray-50 rounded-md grow transition-all"
      style="background-color: gray; color: black;"
    >
      <div
        class="transition-all flex  w-fit items-center justify-center group/letter "
        style="width: 100%; height: 100%;"
      >
        <div
          class="h-[1em] relative overflow-hidden"
        >
          <div
            class="h-[1em] opacity-0"
          >
            tab 4
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
          >
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              4
            </span>
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
          >
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              4
            </span>
          </div>
        </div>
      </div>
    </button>
    <button
      class="p-4 bg-gray-50 rounded-md grow transition-all"
      style="background-color: gray; color: black;"
    >
      <div
        class="transition-all flex  w-fit items-center justify-center group/letter "
        style="width: 100%; height: 100%;"
      >
        <div
          class="h-[1em] relative overflow-hidden"
        >
          <div
            class="h-[1em] opacity-0"
          >
            tab 5
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
          >
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
            >
              5
            </span>
          </div>
          <div
            class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
          >
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              t
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              a
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              b
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
               
            </span>
            <span
              class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
            >
              5
            </span>
          </div>
        </div>
      </div>
    </button>
  </div>
  <main
    class="flex min-h-screen flex-col items-center justify-between p-24"
  >
    <div
      class="transition-all flex  w-fit items-center justify-center group/letter "
      style="height: 100px; font-size: 50px;"
    >
      <div
        class="h-[1em] relative overflow-hidden"
      >
        <div
          class="h-[1em] opacity-0"
        >
          Project
        </div>
        <div
          class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
        >
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            P
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            r
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            o
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            j
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            e
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            c
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            t
          </span>
        </div>
        <div
          class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
        >
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            P
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            r
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            o
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            j
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            e
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            c
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            t
          </span>
        </div>
      </div>
    </div>
    <div
      class="transition-all flex  w-fit items-center justify-center group/letter "
      style="height: 100px; font-size: 50px;"
    >
      <div
        class="h-[1em] relative overflow-hidden"
      >
        <div
          class="h-[1em] opacity-0"
        >
          Project
        </div>
        <div
          class="absolute flex h-[1em] top-[-0.1em] left-0  items-center"
        >
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            P
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            r
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            o
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            j
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            e
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            c
          </span>
          <span
            class=" group-hover/letter:even:animate-letterDown2  group-hover/letter:odd:animate-letterUp2  transition-all ease-in-out  duration-300 [animation-fill-mode:backwards]"
          >
            t
          </span>
        </div>
        <div
          class="absolute flex h-[1em] top-[-0.1em] left-0 items-center"
        >
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            P
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            r
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            o
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            j
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            e
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            c
          </span>
          <span
            class="even:translate-y-[-1em] odd:translate-y-[1em]  group-hover/letter:even:animate-letterDown1  group-hover/letter:odd:animate-letterUp1 transition-all ease-in-out duration-300 [animation-fill-mode:backwards]"
          >
            t
          </span>
        </div>
      </div>
    </div>
  </main>
</div>
`;
